packages/starlight-llms-txt/generator.ts

```diff
@@ -22,11 +22,14 @@ export async function generateLlmsTxt(
 	if (options.minify) {
 		docs = docs.filter((doc) => !micromatch.isMatch(doc.id, starlightLllmsTxtContext.exclude));
 	}
-	docs.sort((a, b) => {
-		const aIsIndex = a.id === 'index';
-		const bIsIndex = b.id === 'index';
-		return aIsIndex && !bIsIndex ? -1 : bIsIndex && !aIsIndex ? 1 : collator.compare(a.id, b.id);
-	});
+	const { promote, demote } = starlightLllmsTxtContext;
+	const prioritizePages = (id: string) => {
+		const demoted = demote.findIndex((expr) => micromatch.isMatch(id, expr));
+		const promoted = demoted > -1 ? -1 : promote.findIndex((expr) => micromatch.isMatch(id, expr));
+		const prefixLength = (promoted > -1 ? promote.length - promoted : 0) + demote.length - demoted - 1;
+		return '_'.repeat(prefixLength) + id;
+	};
```

Comment:
Could we add a comment or two to explain this logic? It’s not mega easy to follow.

Reply:
Sure thing. I tried my best to explain what I'm doing there. Hope that helps a bit. :)